### PR TITLE
Added support for PEP 570 (#867)

### PR DESCRIPTION
**Pull request by Nekit10**
See Issue #867 or https://www.python.org/dev/peps/pep-0570/

### CODE REVIEW

**Comment by asottile** [comment]
fwiw, this patch also fixes the problem, I don't know which one is more correct (there other "unary operators" covering the splat / etc. operations in this list (it's how named-only arguments aren't flagged):

```diff
diff --git a/pycodestyle.py b/pycodestyle.py
index 6ad5456..a0c2898 100755
--- a/pycodestyle.py
+++ b/pycodestyle.py
@@ -112,7 +112,7 @@ REPORT_FORMAT = {
 PyCF_ONLY_AST = 1024
 SINGLETONS = frozenset(['False', 'None', 'True'])
 KEYWORDS = frozenset(keyword.kwlist + ['print', 'async']) - SINGLETONS
-UNARY_OPERATORS = frozenset(['>>', '**', '*', '+', '-'])
+UNARY_OPERATORS = frozenset(['>>', '**', '*', '/', '+', '-'])
 ARITHMETIC_OP = frozenset(['**', '*', '/', '//', '+', '-'])
 WS_OPTIONAL_OPERATORS = ARITHMETIC_OP.union(['^', '&', '|', '<<', '>>', '%'])
 # Warn for -> function annotation operator in py3.5+ (issue 803)
diff --git a/testsuite/python38.py b/testsuite/python38.py
new file mode 100644
index 0000000..2132cd5
--- /dev/null
+++ b/testsuite/python38.py
@@ -0,0 +1,2 @@
+def f(a, /, b):
+    pass
```

**Comment by mikeholler** [comment]
Any updates on this? I'm having issues as well.

**Comment by aviramha** [comment]
> code seems fine, can you make `testsuite/python38.py` with:
> 
> ```python
> def f(a, /, b):
>     pass
> ```

Seems author is stale, should I/anyone else just create a new PR with a test added?

**Comment by Nekit10** [comment]
I will add tests tonight

**Comment by aviramha** [comment]
Any update?

File: testsuite/python38 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#: Okay
```
Comment (asottile) [inline testsuite/python38]: needs to be `python38.py`

File: pycodestyle.py [hunk 2]
```diff
@@ -855,6 +855,10 @@ def missing_whitespace_around_operator(logical_line, tokens):
                 # Tolerate the "<>" operator, even if running Python 3
                 # Deal with Python 3's annotated return value "->"
                 pass
+            elif prev_text == '/' and text == ',':
+                # Tolerate the "/" operator in function definition
+                # For more info see PEP570source
```
Comment (asottile) [inline pycodestyle.py]:
```suggestion
                # For more info see PEP570
```

**Comment by asottile** [comment]
actually, looks like you'll need to rebase now since `python38.py` got created on master

**Comment by Nekit10** [comment]
> this seems fine to me, a smaller patch however:
> 
> ```diff
> -UNARY_OPERATORS = frozenset(['>>', '**', '*', '+', '-'])
> +UNARY_OPERATORS = frozenset(['>>', '**', '*', '/', '+', '-'])
> ```
> 
> (this is how keyword-only arguments are handled currently)
> 
> 
I think it is not a good idea 'cause in this case '/' is not an unary operator

**Comment by asottile** [comment]
> > this seems fine to me, a smaller patch however:
> > ```diff
> > -UNARY_OPERATORS = frozenset(['>>', '**', '*', '+', '-'])
> > +UNARY_OPERATORS = frozenset(['>>', '**', '*', '/', '+', '-'])
> > ```
> > 
> > 
> > (this is how keyword-only arguments are handled currently)
> I think it is not a good idea 'cause in this case '/' is not an unary operator

neither is `>>` nor `*` nor `**`

**Comment by FichteFoll** [comment]
`*` and `**` kinda are unary for unpacking, but they aren't operators in that case. I have no idea why `>>` is in that list, however.

Either way, since * and ** are allowed for unpacking in quite a few situations, it kind of makes sense to have them in that list, but `/` is really only valid in the parameter list and the suggested patch is more targeted and thus makes more sense to me.

**Comment by asottile** [comment]
`>>` is probably for python 2

**Comment by charisz** [comment]
Waiting for release

**Pull request merged**